control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -770,23 +774,36 @@ func (r *reconciler) reconcileIngressController(ctx context.Context, hcp *hyperv
 			errs = append(errs, fmt.Errorf("failed to retrieve guest cluster ingress NodePort: %w", err))
 		}
 
-		hcpNamespace := hcp.Namespace
+		var targetClient client.Client
+		var namespace string
+		// If external infra cluster is being used - use infra client and external cluster namespace
+		// rather than mgmt client and hosted control plane namespace
+		if hcp.Spec.Platform.Kubevirt.ExtClusterKubeConfigSecret != nil &&
+			hcp.Spec.Platform.Kubevirt.ExtClusterNamespace != "" {
+			r.externalInfra = true
+			r.externalInfraNamespace = hcp.Spec.Platform.Kubevirt.ExtClusterNamespace
+			namespace = r.externalInfraNamespace
+			targetClient = r.infraClient
+		} else {
+			targetClient = r.cpClient
+			namespace = hcp.Namespace
+		}
 
 		// Manifests for infra/mgmt cluster passthrough service
-		cpService := manifests.IngressDefaultIngressPassthroughService(hcpNamespace)
+		cpService := manifests.IngressDefaultIngressPassthroughService(namespace)
 
 		// Manifests for infra/mgmt cluster passthrough routes
-		cpPassthroughRoute := manifests.IngressDefaultIngressPassthroughRoute(hcpNamespace)
+		cpPassthroughRoute := manifests.IngressDefaultIngressPassthroughRoute(namespace)
 
-		if _, err := r.CreateOrUpdate(ctx, r.cpClient, cpService, func() error {
+		if _, err := r.CreateOrUpdate(ctx, targetClient, cpService, func() error {
 			return ingress.ReconcileDefaultIngressPassthroughService(cpService, defaultIngressNodePortService, hcp)
-		}); err != nil {
+		}); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
without ignoring the `IsAlreadyExists` error, i'm seeing a reconciliation error in the HCCO logs:
```
{"level":"error","ts":"2023-02-23T14:24:45Z","msg":"Reconciler error","controller":"resources","object":{"name":""},"namespace":"","name":"","reconcileID":"a12f1f45-ad69-4766-bbb3-77a0c9826a73","error":"[failed to reconcile ingress controller: [failed to reconcile kubevirt ingress passthrough service: services \"default-ingress-passthrough-service\" already exists, failed to reconcile kubevirt ingress passthrough route: routes.route.openshift.io \"default-ingress-passthrough-route\" already exists], MutateFn cannot mutate object name and/or object namespace. key: openshift-cluster-csi-drivers/kubevirt-csi-node-cr, kind: &TypeMeta{Kind:ClusterRole,APIVersion:rbac.authorization.k8s.io/v1,}, namespace: ]","errorCauses":[{"error":"failed to reconcile ingress controller: [failed to reconcile kubevirt ingress passthrough service: services \"default-ingress-passthrough-service\" already exists, failed to reconcile kubevirt ingress passthrough route: routes.route.openshift.io \"default-ingress-passthrough-route\" ...
```
and the ingress service & route were created successfully on the infra cluster.

Reply:
This likely has something to do with the logic around caching in the CreateOrUpdate function here [1]. We're wrapping that function with our own custom logic that seems to be tied to the mgmt cluster api somehow.

My guess is that the create or update function is using the mgmt cluster's client cache instead of the kubevirt infra client cache. Maybe the `mgr.GetAPIReader()` here [2] is using the mgmt cluster client which messes up the create or update logic.

I'm not sure exactly, but my guess is that something is off here. We might have to have our own create or update function that's configured for the kubevirt infra client specifically. 

1. https://github.com/openshift/hypershift/blob/42e85b6ed107c4b99a5ef00b886e1943ea896095/control-plane-operator/hostedclusterconfigoperator/cmd.go#L223 
2. https://github.com/openshift/hypershift/blob/42e85b6ed107c4b99a5ef00b886e1943ea896095/control-plane-operator/hostedclusterconfigoperator/cmd.go#L225